ebos/eclmpiserializer.hh

```diff
@@ -27,23 +27,45 @@ namespace Opm {
 
 class EclMpiSerializer {
 public:
+    enum class Operation {
+        PACKSIZE,
+        PACK,
+        UNPACK
+    };
+
     explicit EclMpiSerializer(Dune::CollectiveCommunication<Dune::MPIHelper::MPICommunicator> comm) :
         m_comm(comm)
     {}
 
     template<class T>
-    std::size_t packSize(const T& data) {
-        return Mpi::packSize(data, m_comm);
+    void operator()(const T& data)
+    {
+        if (m_op == Operation::PACKSIZE)
+            m_packSize += Mpi::packSize(const_cast<const T&>(data), m_comm);
+        else if (m_op == Operation::PACK)
+            Mpi::pack(const_cast<const T&>(data), m_buffer, m_position, m_comm);
+        else if (m_op == Operation::UNPACK)
+            Mpi::unpack(const_cast<T&>(data), m_buffer, m_position, m_comm);
```

Comment:
Why do you need `const_cast<const T&>` here?  Is the `data` not already a `const T&`?

Reply:
I don't need them, leftovers from a previous attempt. removed now.